homeassistant/components/adax/climate.py

```diff
@@ -0,0 +1,146 @@
+"""Support for Adax wifi-enabled home heaters."""
+import logging
+
+from adax import Adax
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_PASSWORD,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import ACCOUNT_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
```

Comment:
```suggestion
async def async_setup_entry(
    hass: HomeAssistant,
    entry: ConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
```

Reply:
That would also require typing added to the Adax library?

Reply:
That depends a bit on the place or usage. You can always opt to handle `Any`, cast or ignore a specific typing when interacting with the library. It will at that point still be valuable on this end.

Nevertheless, it isn't a requirement to type.